utils/staking-miner/src/monitor.rs

```diff
@@ -131,39 +209,58 @@ macro_rules! monitor_cmd_for { ($runtime:tt) => { paste::paste! {
 			config: MonitorConfig,
 		) {
 
+			async fn flatten<T>(
```

Comment:
Just out of curiosity, why a function instead of a closue? Is it a limitation of the `async` keyword?

Reply:
async closures are not supported in rust yet, so that's why it's a separate function.

This could be a helper method because `StakingMinerError` needs `T` from the `Runtime` I made it an internal helper.. :(

Reply:
AFAIK `fn` and `const` inside a body of the function are identical to being places in global scope. So, unlike closures, they do not capture anything from their scope.